internal/access/access_key.go

```diff
@@ -20,9 +20,10 @@ func currentAccessKey(c *gin.Context) *models.AccessKey {
 }
 
 func ListAccessKeys(c *gin.Context, identityID uid.ID, name string, showExpired bool, pg models.Pagination) ([]models.AccessKey, error) {
-	db, err := RequireInfraRole(c, models.InfraAdminRole, models.InfraViewRole)
+	roles := []string{models.InfraAdminRole, models.InfraViewRole}
+	db, err := RequireInfraRole(c, roles...)
 	if err != nil {
-		return nil, err
+		return nil, HandleAuthErr(err, "access keys", "list", roles...)
```

Comment:
I think ideally instead of a bunch of separate methods in `access` we would have a single method that accepted a struct, and returned the `AuthorizationError`. And our data access would be either directly in the API handler, or some other function. Something like this:

```go

type Request struct {
    Resource string
    Operation string
    RequiredRoles []string
}

func Authorize(req Request) error {
    ...
    return AuthorizationError{...}
}

```

That change would be much too large to make right now, so we're forced to keep the `RequireInfraRole` and `HandleAuthErr` calls in-sync. That shouldn't be too bad because they should always be just a couple lines apart in the same function.

Reply:
Love this idea to consolidate the layers here. It would remove quite a bit of bulk.